src/internal/common/idname/idname.go

```diff
@@ -0,0 +1,89 @@
+package idname
+
+import (
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type Provider interface {
+	// the canonical id of the thing, generated and usable
+	//  by whichever system has ownership of it.
+	ID() string
+	// the human-readable name of the thing.
+	Name() string
+}
+
+var _ Provider = &Is{}
+
+// Is provides an id-name tuple.
+type Is struct {
+	IDV   string
+	NameV string
+}
```

Comment:
this doesn't seem to be used outside this package so it and all its fields should be unexported

Reply:
ah, had a moment where it did need an export.  Good catch.